### PR TITLE
ratfor (new formula)

### PR DESCRIPTION
**Pull request by FranklinChen**

### CODE REVIEW

File: Library/Formula/ratfor.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Ratfor < Formula
+  version "1.02"
+  desc "Rational Fortran"
+  homepage "http://www.dgate.org/ratfor/"
+  url "http://www.dgate.org/ratfor/ratfor.tar.gz"
```
Comment (DomT4) [inline Library/Formula/ratfor.rb]: We don't really use static URLs in the core. It needs to be a versioned link that won't silently break on us.

File: Library/Formula/ratfor.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  sha256 "daf2971df48c3b3908c6788c4c6b3cdfb4eaad21ec819eee70a060956736ea1c"
+
+  depends_on "gcc"
```
Comment (DomT4) [inline Library/Formula/ratfor.rb]: For build?

File: Library/Formula/ratfor.rb (hunk 1 above)
Comment (FranklinChen) [inline Library/Formula/ratfor.rb]: It's the only URL provided.

File: Library/Formula/ratfor.rb (hunk 2 above)
Comment (FranklinChen) [inline Library/Formula/ratfor.rb]: No, for use of the Fortran code generated. I guess that should be noted rather than a requirement for installation.

Comment (FranklinChen) [inline Library/Formula/ratfor.rb]: "make check" uses gfortran from gcc.

File: Library/Formula/ratfor.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/ratfor.rb]: You'll need to ask upstream for a versioned one, or host a tap for this yourself. Static URLs are bad UX, sadly.

File: Library/Formula/ratfor.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/ratfor.rb]: `depends_on :fortran` then?

**Comment by dunn** [comment]
A few audit failures:
```
==> audit problems
ratfor:
 * `url` (line 5) should be put before `version` (line 2)
 * A `test do` test block should be added
 * Stable: version 1.02 is redundant with version scanned from URL
```
Upshot is you can remove the `version` line entirely, and we'd like a test that checks basic functionality.

**Comment by FranklinChen** [comment]
How do I move "make check" from the install phase to the test block?

**Comment by dunn** [comment]
Afraid that won't work; we run the test in a separate temporary directory so it doesn't have access to the source code.  Is there a command that can be run non-interactively and won't take too long?  CMake is a good example: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/cmake.rb#L125-L128

**Comment by DomT4** [comment]
Closing in favour of #44358.

**Pull request closed without merging**